### PR TITLE
kgo: make api reference versioned

### PR DESCRIPTION
**Pull request by pmalek**
### Description

This will allow us to version the api reference.

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_data/docs_nav_kgo_1.0.x.yml [hunk 1]
```diff
@@ -79,12 +79,13 @@ items:
       - text: Custom Resources
         items:
           - text: Overview
-            url: /reference/custom-resources/
+            url: /reference/custom-resources/api-reference
+            src: reference/custom-resources/api-reference-1.0.x
           - text: GatewayConfiguration
-            url: /reference/custom-resources/#gatewayconfiguration
+            url: /reference/custom-resources/api-reference-v1.0.md#gatewayconfiguration
```
Comment (acgoldsmith) [inline app/_data/docs_nav_kgo_1.0.x.yml:85]:
```suggestion
            url: /reference/custom-resources/api-reference-v1.0/#gatewayconfiguration
```

File: app/_data/docs_nav_kgo_1.0.x.yml [hunk 2]
```diff
@@ -79,12 +79,13 @@ items:
...
           - text: ControlPlane
-            url: /reference/custom-resources/#controlplane
+            url: /reference/custom-resources/api-reference-v1.0.md#controlplane
```
Comment (acgoldsmith) [inline app/_data/docs_nav_kgo_1.0.x.yml:87]:
```suggestion
            url: /reference/custom-resources/api-reference-v1.0/#controlplane
```

File: app/_data/docs_nav_kgo_1.0.x.yml [hunk 3]
```diff
@@ -79,12 +79,13 @@ items:
...
           - text: DataPlane
-            url: /reference/custom-resources/#dataplane
+            url: /reference/custom-resources/api-reference-v1.0.md#dataplane
```
Comment (acgoldsmith) [inline app/_data/docs_nav_kgo_1.0.x.yml:89]:
```suggestion
            url: /reference/custom-resources/api-reference-v1.0/#dataplane
```

File: app/_data/docs_nav_kgo_1.0.x.yml (hunk 1 above)
Comment (mheap) [inline app/_data/docs_nav_kgo_1.0.x.yml:85]:
```suggestion
            url: /reference/custom-resources/api-reference#gatewayconfiguration
            src: reference/custom-resources/api-reference-1.0.x
```

**Review by acgoldsmith — COMMENTED** [review]
@pmalek , I think if we replace the .md w / that should solve...

**Comment by pmalek** [comment]
Last versino https://github.com/Kong/docs.konghq.com/pull/6531/commits/9cb85150f3e9941665f3249945b5f8caf1c00a65 from @mheap works 🥳

**Review by mheap — APPROVED** [review]

**Pull request merged**